model/task_lifecycle.go

```diff
@@ -1418,20 +1418,8 @@ func ClearAndResetStrandedTask(h *host.Host) error {
 		return nil
 	}
 
-	if err = h.ClearRunningTask(); err != nil {
```

Comment:
This being out of order isn't relevant to the bug I found but does still appear to be a bug.

Reply:
Just to check - the ordering is for the same reasoning about blocking single-host task groups as in #5640, right?

Reply:
Yes exactly 👍